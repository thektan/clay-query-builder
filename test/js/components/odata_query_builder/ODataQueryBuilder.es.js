import React from 'react';
import ODataQueryBuilder from 'components/odata_query_builder/ODataQueryBuilder.es';
import {cleanup, render} from 'react-testing-library';

const QUERY_INPUT_TESTID = 'query-input';

describe(
	'ODataQueryBuilder',
	() => {
		afterEach(cleanup);

		it(
			'should render with no properties',
			() => {
				const {asFragment} = render(
					<ODataQueryBuilder properties={[]} />
				);

				expect(asFragment()).toMatchSnapshot();
			}
		);

		it(
			'should render with properties',
			() => {
				const {asFragment} = render(
					<ODataQueryBuilder
						properties={[
							{
								label: 'First Name',
								name: 'firstName',
								type: 'string'
							},
							{
								label: 'Last Name',
								name: 'lastName',
								type: 'string'
							},
						]}
					/>
				);

				expect(asFragment()).toMatchSnapshot();
			}
		);

		it(
			'should populate the query input with a valid initial query',
			() => {
				const {getByTestId} = render(
					<ODataQueryBuilder
						properties={[
							{
								label: 'First Name',
								name: 'firstName',
								type: 'string'
							}
						]}
						initialQuery={`(firstName eq 'test')`}
					/>
				);

				const queryInput = getByTestId(QUERY_INPUT_TESTID);

				expect(queryInput.value).toEqual(`(firstName eq 'test')`);
			}
		);

		it(
			'should not break if the property in the initial query does not exist in the accepted properties',
			() => {
				const {getByTestId} = render(
					<ODataQueryBuilder
						properties={[]}
						initialQuery={`(firstName eq 'test')`}
					/>
				);

				const queryInput = getByTestId(QUERY_INPUT_TESTID);

				expect(queryInput.value).toEqual(`(firstName eq 'test')`);
			}
		);

		it(
			'should render a blank input if initial query is invalid',
			() => {
				const {getByTestId} = render(
					<ODataQueryBuilder
						properties={[]}
						initialQuery={`(test)`}
					/>
				);

				const queryInput = getByTestId(QUERY_INPUT_TESTID);

				expect(queryInput.value).toEqual('');
			}
		);
	}
);