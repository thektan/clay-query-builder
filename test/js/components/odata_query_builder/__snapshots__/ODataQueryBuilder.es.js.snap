// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ODataQueryBuilder should render with no properties 1`] = `
<DocumentFragment>
  <div
    class="clay-query-builder-root"
  >
    <div
      class="form-group"
    >
      <div
        class="criteria-builder-root read-only"
      >
        <div
          class="criteria-builder-section-main"
        >
          <div
            class="criteria-builder-toolbar"
          >
            <label
              class="toggle-switch"
            >
              <input
                class="toggle-switch-check"
                type="checkbox"
              />
              <span
                aria-hidden="true"
                class="toggle-switch-bar"
              >
                <span
                  class="toggle-switch-handle"
                  data-label-off=""
                  data-label-on=""
                >
                  <span
                    class="button-icon button-icon-off toggle-switch-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="lexicon-icon lexicon-icon-pencil"
                      viewBox="0 0 512 512"
                    >
                      <use
                        xlink:href="#pencil"
                      />
                    </svg>
                  </span>
                  <span
                    class="button-icon button-icon-on toggle-switch-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="lexicon-icon lexicon-icon-pencil"
                      viewBox="0 0 512 512"
                    >
                      <use
                        xlink:href="#pencil"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </label>
          </div>
          <div
            class="criteria-group-root criteria-group-item-root"
          >
            <div
              class="empty-drop-zone-root"
            >
              <div
                class="empty-drop-zone-target"
              >
                <div
                  class="empty-drop-zone-indicator"
                />
                <div
                  class="empty-drop-zone-help-message"
                >
                  <div
                    class="message-item"
                  >
                    <img
                      class="message-icon"
                      src="/drag-and-drop.svg"
                    />
                    <span>
                      drag-and-drop-criterion-from-the-right-to-add-rules
                    </span>
                  </div>
                  <div
                    class="message-item"
                  >
                    <img
                      class="message-icon"
                      src="/drag-over.svg"
                    />
                    <span>
                      drag-and-drop-over-an-existing-criteria-to-form-groups
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="criteria-builder-section-sidebar"
        >
          <div
            class="criteria-sidebar-root"
          >
            <div
              class="sidebar-header"
            >
              x-properties
            </div>
            <div
              class="sidebar-search"
            >
              <div
                class="input-group"
              >
                <div
                  class="input-group-item"
                >
                  <input
                    class="form-control input-group-inset input-group-inset-after"
                    data-testid="search-input"
                    placeholder="search"
                    type="text"
                    value=""
                  />
                  <div
                    class="input-group-inset-item input-group-inset-item-after"
                  >
                    <button
                      class="btn btn-unstyled"
                      data-testid="search-button"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="lexicon-icon lexicon-icon-search icon"
                        viewBox="0 0 512 512"
                      >
                        <use
                          xlink:href="#search"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <ul
              class="properties-list"
            >
              <li
                class="empty-message"
              >
                no-results-were-found
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      class="form-group"
    >
      <textarea
        class="field form-control"
        data-testid="query-input"
        readonly=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ODataQueryBuilder should render with properties 1`] = `
<DocumentFragment>
  <div
    class="clay-query-builder-root"
  >
    <div
      class="form-group"
    >
      <div
        class="criteria-builder-root read-only"
      >
        <div
          class="criteria-builder-section-main"
        >
          <div
            class="criteria-builder-toolbar"
          >
            <label
              class="toggle-switch"
            >
              <input
                class="toggle-switch-check"
                type="checkbox"
              />
              <span
                aria-hidden="true"
                class="toggle-switch-bar"
              >
                <span
                  class="toggle-switch-handle"
                  data-label-off=""
                  data-label-on=""
                >
                  <span
                    class="button-icon button-icon-off toggle-switch-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="lexicon-icon lexicon-icon-pencil"
                      viewBox="0 0 512 512"
                    >
                      <use
                        xlink:href="#pencil"
                      />
                    </svg>
                  </span>
                  <span
                    class="button-icon button-icon-on toggle-switch-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="lexicon-icon lexicon-icon-pencil"
                      viewBox="0 0 512 512"
                    >
                      <use
                        xlink:href="#pencil"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </label>
          </div>
          <div
            class="criteria-group-root criteria-group-item-root"
          >
            <div
              class="empty-drop-zone-root"
            >
              <div
                class="empty-drop-zone-target"
              >
                <div
                  class="empty-drop-zone-indicator"
                />
                <div
                  class="empty-drop-zone-help-message"
                >
                  <div
                    class="message-item"
                  >
                    <img
                      class="message-icon"
                      src="/drag-and-drop.svg"
                    />
                    <span>
                      drag-and-drop-criterion-from-the-right-to-add-rules
                    </span>
                  </div>
                  <div
                    class="message-item"
                  >
                    <img
                      class="message-icon"
                      src="/drag-over.svg"
                    />
                    <span>
                      drag-and-drop-over-an-existing-criteria-to-form-groups
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="criteria-builder-section-sidebar"
        >
          <div
            class="criteria-sidebar-root"
          >
            <div
              class="sidebar-header"
            >
              x-properties
            </div>
            <div
              class="sidebar-search"
            >
              <div
                class="input-group"
              >
                <div
                  class="input-group-item"
                >
                  <input
                    class="form-control input-group-inset input-group-inset-after"
                    data-testid="search-input"
                    placeholder="search"
                    type="text"
                    value=""
                  />
                  <div
                    class="input-group-inset-item input-group-inset-item-after"
                  >
                    <button
                      class="btn btn-unstyled"
                      data-testid="search-button"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="lexicon-icon lexicon-icon-search icon"
                        viewBox="0 0 512 512"
                      >
                        <use
                          xlink:href="#search"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <ul
              class="properties-list"
            >
              <li
                class="criteria-sidebar-item-root"
                draggable="true"
              >
                <span
                  class="inline-item"
                >
                  <svg
                    aria-hidden="true"
                    class="lexicon-icon lexicon-icon-drag"
                    viewBox="0 0 512 512"
                  >
                    <use
                      xlink:href="#drag"
                    />
                  </svg>
                </span>
                <span
                  class="criteria-sidebar-item-type sticker sticker-light"
                >
                  <span
                    class="inline-item"
                  >
                    <svg
                      aria-hidden="true"
                      class="lexicon-icon lexicon-icon-text"
                      viewBox="0 0 512 512"
                    >
                      <use
                        xlink:href="#text"
                      />
                    </svg>
                  </span>
                </span>
                First Name
              </li>
              <li
                class="criteria-sidebar-item-root"
                draggable="true"
              >
                <span
                  class="inline-item"
                >
                  <svg
                    aria-hidden="true"
                    class="lexicon-icon lexicon-icon-drag"
                    viewBox="0 0 512 512"
                  >
                    <use
                      xlink:href="#drag"
                    />
                  </svg>
                </span>
                <span
                  class="criteria-sidebar-item-type sticker sticker-light"
                >
                  <span
                    class="inline-item"
                  >
                    <svg
                      aria-hidden="true"
                      class="lexicon-icon lexicon-icon-text"
                      viewBox="0 0 512 512"
                    >
                      <use
                        xlink:href="#text"
                      />
                    </svg>
                  </span>
                </span>
                Last Name
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      class="form-group"
    >
      <textarea
        class="field form-control"
        data-testid="query-input"
        readonly=""
      />
    </div>
  </div>
</DocumentFragment>
`;
